src/ResultWriter/WaveFieldWriter.cpp

```diff
@@ -477,19 +507,21 @@ void seissol::writer::WaveFieldWriter::write(double time) {
   // nextId is required in a manner similar to above for writing integrated variables
   nextId = 0;
 
-  if (m_integrals) {
-    for (unsigned int i = 0; i < WaveFieldWriterExecutor::NUM_INTEGRATED_VARIABLES; i++) {
-      if (!m_lowOutputFlags[i])
+  if (m_integrals != nullptr) {
+    for (unsigned int i = 0; i < WaveFieldWriterExecutor::NumIntegratedVariables; i++) {
+      if (!m_lowOutputFlags[i]) {
         continue;
+      }
       real* managedBuffer =
           async::Module<WaveFieldWriterExecutor, WaveFieldInitParam, WaveFieldParam>::managedBuffer<
               real*>(m_variableBufferIds[1] + nextId);
 
 #ifdef _OPENMP
 #pragma omp parallel for schedule(static)
 #endif // _OPENMP
-      for (unsigned int j = 0; j < m_numLowCells; j++)
+      for (unsigned int j = 0; j < m_numLowCells; j++) {
```

Comment:
Ok, can do.

Reply:
On second thought; maybe it may be good to wait a bit. We may get rid of those files sooner than later anyways (#1180), so maybe it's also ok right now? If #1180 takes too long, we can still do it in a follow-up cleanup PR. Would that be good?